lib/Room.php

```diff
@@ -284,6 +321,10 @@ public function enterRoomAsUser($userId) {
 		$result = $query->execute();
 
 		if ($result === 0) {
+			if ($this->getPassword() !== '' && $this->getPassword() !== $password) {
```

Comment:
Use `hash_equals` instead of `!==` to compare secret with user-provided value.

Reply:
Well we don't compare hashes yes. Also see the note in docs:
http://php.net/manual/en/function.hash-equals.php
> Both arguments must be of the same length to be compared successfully. When arguments of differing length are supplied, FALSE is returned immediately and the length of the known string may be leaked in case of a timing attack. 

Also I'm pretty sure you can not do timing attacks against nextcloud apis 🙈

Reply:
`hash_equals` can be used to compare _any_ kind of strings, be it hashes or PINs.
Leaking the length of the secret is fine, it is specified in this open source code after all.

> Also I'm pretty sure you can not do timing attacks against nextcloud apis

I wouldn't bet on that.

Reply:
> Leaking the length of the secret is fine, it is specified in this open source code after all.

1. There is no secret neither PINs, we have a room password here
2. The length is unknown, since it will be provided by the owner of the room
3. It's not specified in any code Oo?

Don't get me wrong, we will do our best to make this more secure in the end after the basic functionality works, but there seem to be a couple of misunderstandings here.

Reply:
It's now using OCP\Security\IHasher